tests/data/config_broken_dimensions.yaml

```diff
@@ -0,0 +1,29 @@
+billing_interval: monthly
+product_code: foo
+query_interval: 3600
+reporting_api_is_cumulative: true
+reporting_interval: 3600
+usage_metrics:
+  managed_node_count:
+    consumption_reporting: volume
+    dimensions:
+    - dimension: tier_1
+      max: 15
+      min: 0
+    - dimension: tier_2
+      max: 50
+      min: 16
+    - dimension: tier_3
+      max: 100
+      min: 51
+    - dimension: tier_4
+      max: 250
+      min: 101
+    - dimension: tier_5
+      max: 500
+      min: 251
+    - dimension: tier_6
+      min: 1001
```

Comment:
but this is not broken, it would be broken if `tier_6` had a `max` setting. For example if `max: 1500` and the count we are trying to match is `1501` then the algorithm would throw the new exception. I do not think the tests reflect this condition accurately. So for the test the broken config could simply end after `tier_2` and then the test could run with a simulated node count of `18`

Reply:
Ok will do...

Reply:
I have updated the tests to add an additional fall off the end test, as well as the existing gap test, and have renamed the config files, and re-ordered the min/max entries to make it easier to read, and added comments indicating the nature of the broken dimensions.